packages/ethereum-contracts/contracts/apps/TCILibraryV1.sol

```diff
@@ -0,0 +1,190 @@
+
+    function setFlowPermissions(
+        ISuperToken token,
+        address flowOperator,
+        bool allowCreate,
+        bool allowUpdate,
+        bool allowDelete,
+        int96 flowRateAllowance
+    ) internal returns (bool) {
+        (ISuperfluid host, IConstantFlowAgreementV1 cfa) = _getAndCacheHostAndCFA(token);
+        uint8 permissionsBitmask = (allowCreate ? 1 : 0)
+            | (allowUpdate ? 1 : 0) << 1
+            | (allowDelete ? 1 : 0) << 2;
+        host.callAgreement(
+            cfa,
+            abi.encodeCall(
+                cfa.updateFlowOperatorPermissions,
+                (token, flowOperator, permissionsBitmask, flowRateAllowance, new bytes(0))
+            ),
+            new bytes(0)
+        );
+        return true;
+    }
+
+    function getFlowRate(ISuperToken token, address sender, address receiver)
+        internal view returns(int96 flowRate)
+    {
+        (, IConstantFlowAgreementV1 cfa) = _getHostAndCFA(token);
+        (, flowRate, , ) = cfa.getFlow(token, sender, receiver);
+    }
+
+    function getFlowInfo(ISuperToken token, address sender, address receiver)
+        internal view
+        returns(uint256 lastUpdated, int96 flowRate, uint256 deposit, uint256 owedDeposit)
+    {
+        (, IConstantFlowAgreementV1 cfa) = _getHostAndCFA(token);
+        (lastUpdated, flowRate, deposit, owedDeposit) = cfa.getFlow(token, sender, receiver);
+    }
+
+    function getNetFlowRate(ISuperToken token, address account)
+        internal view returns (int96 flowRate)
+    {
+        (, IConstantFlowAgreementV1 cfa) = _getHostAndCFA(token);
+        return cfa.getNetFlow(token, account);
+    }
+
+    function getNetFlowInfo(ISuperToken token, address account)
+        internal view
+        returns (uint256 lastUpdated, int96 flowRate, uint256 deposit, uint256 owedDeposit)
+    {
+        (, IConstantFlowAgreementV1 cfa) = _getHostAndCFA(token);
+        return cfa.getAccountFlowInfo(token, account);
+    }
+
+    function getBufferAmountByFlowrate(ISuperToken token, int96 flowRate) external view
+        returns (uint256 bufferAmount)
+    {
+        (, IConstantFlowAgreementV1 cfa) = _getHostAndCFA(token);
+        return cfa.getDepositRequiredForFlowRate(token, flowRate);
+    }
+
+    function getFlowPermissions(ISuperToken token, address sender, address flowOperator)
+        internal view
+        returns (bool allowCreate, bool allowUpdate, bool allowDelete, int96 flowRateAllowance)
+    {
+        (, IConstantFlowAgreementV1 cfa) = _getHostAndCFA(token);
+        uint8 permissionsBitmask;
+        (, permissionsBitmask, flowRateAllowance) = cfa.getFlowOperatorData(token, sender, flowOperator);
+        allowCreate = permissionsBitmask & 1 == 1 ? true : false;
+        allowUpdate = permissionsBitmask >> 1 & 1 == 1 ? true : false;
+        allowDelete = permissionsBitmask >> 2 & 1 == 1 ? true : false;
+    }
+
+    // ************** helpers **************
+
+    // keccak256("Superfluid.v1")
```

Comment:
Looks good, just to give a bit more thinking re storage slot conflict possibility considering [solidity storage state layout rules](https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html)

Make the hashed string also a bit more specific (java path style, e.g.)

Reply:
Makes sens re: @hellwolf